nbgrader/apps/zipcollectapp.py

```diff
@@ -0,0 +1,522 @@
+    ),
+    'update-db': (
+        {'ZipCollectApp' : {'auto_update_database' : True}},
+        "Automatically update the database."
+    ),
+}
+
+
+class ZipCollectApp(NbGrader):
+
+    name = u'nbgrader-zip-collect'
+    description = u'Collect assignments from archives (zip files).'
+
+    aliases = aliases
+    flags = flags
+
+    examples = """
+        Collect assignment submissions from files and/or archives (zip) files
+        manually downloaded from a LMS. For the usage of instructors.
+
+        This command is run from the top-level nbgrader folder. In order to
+        facilitate the collect process, nbgrader places some constraints on how
+        the manually downloaded archive (zip) files must be structured. By
+        default, the directory structure must look like this:
+
+            {downloaded}/{assignment_id}/{collect_step}
+
+        where `downloaded` is the main directory, `assignment_id` is the name
+        of the assignment and `collect_step` is the step in the collect
+        process.
+
+        Manually downloaded assignment submissions files and/or archives (zip)
+        files must be placed in the `archive_directory`:
+
+            {downloaded}/{assignment_id}/{archive_directory}
+
+        It is expected that the instructor has already created this directory
+        and placed the downloaded assignment submissions files and/or archives
+        (zip) files in this directory.
+
+        Archive (zip) files in the `archive_directory` will be extracted, and
+        any non-archive files will be copied, to the `extracted_directory`:
+
+            {downloaded}/{assignment_id}/{extracted_directory}
+
+        After which the files in the `extracted_directory` will be collected
+        and copied into the students `submitted_directory`:
+
+            {submitted_directory}/{student_id}/{assignment_id}/{notebook_id}.ipynb
+
+        By default the collection of files in the `extracted_directory` is
+        managed via the :class:`~nbgrader.plugins.zipcollect.FileNameProcessor`
+        plugin. Each filename is sent to the plugin, which in turn returns an
+        object containing the `student_id`, `file_id`, `first_name`,
+        `last_name`, `email`, and `timestamp` data. For more information run:
+
+            nbgrader zip_collect --help-all
+
+        To change the default plugin, you will need a class that inherits from
+        :class:`~nbgrader.plugins.zipcollect.FileNameProcessor`. If your
+        exporter is named `MyCustomProcessor` and is saved in the file
+        `myprocessor.py`, then:
+
+            nbgrader zip_collect --processor=myprocessor.MyCustomProcessor
+
+        """
+
+    force = Bool(
+        default_value=False,
+        help="Force overwrite of existing files."
+    ).tag(config=True)
+
+    strict = Bool(
+        default_value=False,
+        help="Skip submitted notebooks with invalid names."
+    ).tag(config=True)
+
+    auto_update_database = Bool(
```

Comment:
This option seems like it is redundant with `--create` in `nbgrader autograde`.

... which I realize is technically deprecated. I opened #668 to fix this. But I think this is probably an option that is more robust to have in  `nbgrader autograde` rather than here.